From b87c913695c90af199b2335bdc5d117f658cd132 Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Wed, 25 Apr 2012 21:23:21 -0400
Subject: [PATCH] Copy lock messages before swapping bytes because we
 may use the lock message struct again after calling
 send_lock_msg()

Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
Reviewed-by: Lon Hohberger <lhh@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
---
 rgmanager/src/daemons/cpglockd.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/rgmanager/src/daemons/cpglockd.c b/rgmanager/src/daemons/cpglockd.c
index df18eb3..fa40e98 100644
--- a/rgmanager/src/daemons/cpglockd.c
+++ b/rgmanager/src/daemons/cpglockd.c
@@ -427,11 +427,13 @@ send_lock_msg(struct cpg_lock_msg *m)
 {
 	struct iovec iov;
 	int ret;
+	struct cpg_lock_msg out_msg;
 
-	swab_cpg_lock_msg(m);
+	memcpy(&out_msg, m, sizeof(out_msg));
+	swab_cpg_lock_msg(&out_msg);
 
-	iov.iov_base = m;
-	iov.iov_len = sizeof (*m);
+	iov.iov_base = &out_msg;
+	iov.iov_len = sizeof(out_msg);
 
 	do {
 		ret = cpg_mcast_joined(cpg, CPG_TYPE_AGREED, &iov, 1);
-- 
1.7.7.6

